ddl/db_test.go

```diff
@@ -1988,6 +1991,80 @@ func (s *testDBSuite1) TestCreateTable(c *C) {
 	c.Assert(err.Error(), Equals, "[types:1291]Column 'a' has duplicated value 'B' in ENUM")
 }
 
+func (s *testDBSuite6) TestRepairTable(c *C) {
+	s.tk.MustExec("use test")
+	// Set the repair config.
+	turnRepairModeAndInit(true)
+	defer turnRepairModeAndInit(false)
+
+	// Domain reload the tableInfo and add it into repairInfo.
+	s.tk.MustExec("CREATE TABLE origin (a int primary key, b varchar(10));")
+	dom := domain.GetDomain(s.s)
+	// Make sure the table schema is latest.
+	err := dom.Reload()
+	c.Assert(err, IsNil)
+	// Repaired tableInfo has been filtered by `domain.InfoSchema()`, so get it in repairInfo.
+	originTableInfo := domainutil.RepairInfo.GetRepairedTableInfoByTableName("test", "origin")
+
+	hook := &ddl.TestDDLCallback{}
+	var repairErr error
+	hook.OnJobRunBeforeExported = func(job *model.Job) {
+		if job.Type != model.ActionRepairTable {
+			return
+		}
+		if job.TableID != originTableInfo.ID {
+			repairErr = errors.New("table id should be the same")
+			return
+		}
+		if job.SchemaState != model.StateNone {
+			repairErr = errors.New("repair job state should be the none")
+			return
+		}
+		// Test whether it's readable, when repaired table is still stateNone.
+		tkInternal := testkit.NewTestKitWithInit(c, s.store)
+		_, repairErr = tkInternal.Exec("select * from origin")
+		// Repaired tableInfo has been filtered by `domain.InfoSchema()`, here will get an error cause user can't get access to it.
+		if repairErr != nil && repairErr.Error() == "[schema:1146]Table 'test.origin' doesn't exist" {
```

Comment:
Could we use `terror.ErrorEqual` directly?

Reply:
Good, Addressed